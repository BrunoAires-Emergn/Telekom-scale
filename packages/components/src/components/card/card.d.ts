import { Element } from '@stencil/core';
export declare class Card {
    hostElement: HTMLElement;
    size?: string;
    theme?: string;
    variant?: string;
    disabled?: boolean;
    deselected?: boolean;
    imageTop?: string;
    imageTopAlt?: string;
    private hasSlotHeader;
    private hasSlotFooter;
    private getCssClassMap;
    componentWillLoad(): void;
    render(): any;
}
