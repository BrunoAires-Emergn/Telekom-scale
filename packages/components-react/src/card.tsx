import * as React from 'react'
require('@telements/styles/dist/card.css');

export const Card: React.SFC<{}> = ({ children }) => (
	<div className="card">
		<div className="card__body">
			{children}
		</div>
	</div>
)
