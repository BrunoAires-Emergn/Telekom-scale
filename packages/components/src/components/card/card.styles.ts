import { JssStyle } from 'jss';

export const styles: JssStyle = {
  card: {
    boxSizing: 'border-box',
    overflow: 'hidden',
    color: '#333',
    background: '#fff',
    border: '1px solid rgba(0, 0, 0, 0.1)',
    borderRadius: '4px',
    width: '100%',
    maxWidth: '400px',
    boxShadow: '0 2px 4px 0 rgba(0, 0, 0, 0.12)',
  },
  card__body: {
    padding: '1rem',
  },
  card__header: {
    padding: '0.5rem 1rem',
    borderBottom: '1px solid #dfdfdf',
    background: '#eee',
    fontSize: '0.8rem',
    '& ::slotted(*)': {
      margin: 0,
    },
  },
  card__footer: {
    padding: '0.5rem 1rem',
    borderTop: '1px solid #dfdfdf',
    background: '#eee',
    fontSize: '0.8rem',
    '& ::slotted(*)': {
      margin: 0,
    },
  },
};
