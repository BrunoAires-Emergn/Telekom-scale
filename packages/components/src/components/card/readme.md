# t-card



<!-- Auto Generated Below -->


## Properties

| Property      | Attribute       | Description                                  | Type                                     | Default     |
| ------------- | --------------- | -------------------------------------------- | ---------------------------------------- | ----------- |
| `customClass` | `custom-class`  | (optional) Card class                        | `string`                                 | `''`        |
| `imageTop`    | `image-top`     | (optional) Card image at the top             | `string`                                 | `undefined` |
| `imageTopAlt` | `image-top-alt` | (optional) Card image alternative at the top | `string`                                 | `''`        |
| `size`        | `size`          | (optional) Card size                         | `string`                                 | `''`        |
| `styles`      | --              | (optional) Injected jss styles               | `StyleSheet<string \| number \| symbol>` | `undefined` |
| `variant`     | `variant`       | (optional) Card variant                      | `string`                                 | `''`        |


## Methods

### `updateStyles(newStyle: StyleSheet<string | number | symbol>) => Promise<void>`

Method: updateStyles()

#### Returns

Type: `Promise<void>`




----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
