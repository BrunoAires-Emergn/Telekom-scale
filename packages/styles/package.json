{
    "name": "@telements/styles",
    "version": "0.0.1",
    "main": "dist/index.js",
    "scripts": {
        "build": "tsc -b",
        "postbuild": "node ./dist/export-css.js",
        "setup": "yarn"
    },
    "author": "@eeegor <egor.kirpichev@telekom.de>",
    "license": "MIT",
    "devDependencies": {
        "typescript": "^3.4.5"
    },
    "dependencies": {

    }
}
