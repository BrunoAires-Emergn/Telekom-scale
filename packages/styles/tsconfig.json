{
	"extends": "../tsconfig.base",
	"compilerOptions": {
		"rootDir": "./src",
		"outDir": "./dist",
		"types": [
			"node",
			"jest"
		]
	}
}
