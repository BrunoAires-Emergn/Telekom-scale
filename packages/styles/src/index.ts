import { button } from './button';
import { alert } from './alert';
import { card } from './card';

const theme = {
	colors: {
		white: '#fff',
		black: '#000',
		blue: '#00f',
		red: '#f00',
	}
}

export {
	theme,
	alert,
	button,
	card,
}
