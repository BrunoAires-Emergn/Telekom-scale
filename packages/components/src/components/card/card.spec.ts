import { newSpecPage } from '@stencil/core/testing';
import { Card } from './card';
import { styles } from './card.styles';
import jss from 'jss';

describe('Card', () => {
  let element;
  let stylesheet;
  beforeEach(async () => {
    element = new Card();
    stylesheet = element.stylesheet = jss.createStyleSheet(styles as any);
  });

  it('should match snapshot', async () => {
    const page = await newSpecPage({
      components: [Card],
      html: `<t-card>Label</t-card>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('should match snapshot with header slot', async () => {
    const page = await newSpecPage({
      components: [Card],
      html: `
			<t-card>
				<h3 slot="header">Header content</h3>
				A title
			</t-card>
			`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('should match snapshot with footer slot', async () => {
    const page = await newSpecPage({
      components: [Card],
      html: `
			<t-card>
				<h3 slot="footer">Footer content</h3>
				A title
			</t-card>
			`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('should match snapshot with image', async () => {
    const page = await newSpecPage({
      components: [Card],
      html: `<t-card image-top="http://placehold.it/400x300">A title</t-card>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('should handle css classes', () => {
    element.customClass = 'custom';
    expect(element.getCssClassMap()).toContain('custom');

    element.size = 'small';
    stylesheet.addRule('card--size-small', {});
    expect(element.getCssClassMap()).toContain(
      stylesheet.classes['card--size-small']
    );

    element.variant = 'primary';
    stylesheet.addRule('card--variant-primary', {});
    expect(element.getCssClassMap()).toContain(
      stylesheet.classes['card--variant-primary']
    );
  });
});
