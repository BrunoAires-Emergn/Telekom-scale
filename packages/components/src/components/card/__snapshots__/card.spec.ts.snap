// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match snapshot 1`] = `
<scale-card>
  <mock:shadow-root>
    <style>
      .card-0-0-5 {
  color: #333;
  width: 100%;
  border: 1px solid rgba(0, 0, 0, 0.1);
  overflow: hidden;
  max-width: 400px;
  background: #fff;
  box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.12);
  box-sizing: border-box;
  border-radius: 4px;
}
.card__body-0-0-6 {
  padding: 1rem;
}
.card__header-0-0-7 {
  padding: 0.5rem 1rem;
  font-size: 0.8rem;
  background: #eee;
  border-bottom: 1px solid #dfdfdf;
}
.card__header-0-0-7 ::slotted(*) {
  margin: 0;
}
.card__footer-0-0-8 {
  padding: 0.5rem 1rem;
  font-size: 0.8rem;
  background: #eee;
  border-top: 1px solid #dfdfdf;
}
.card__footer-0-0-8 ::slotted(*) {
  margin: 0;
}
    </style>
    <div class="card-0-0-5">
      <div class="card__body-0-0-6">
        <slot></slot>
      </div>
    </div>
  </mock:shadow-root>
  Label
</scale-card>
`;

exports[`Card should match snapshot with footer slot 1`] = `
<scale-card>
  <mock:shadow-root>
    <style>
      .card-0-0-21 {
  color: #333;
  width: 100%;
  border: 1px solid rgba(0, 0, 0, 0.1);
  overflow: hidden;
  max-width: 400px;
  background: #fff;
  box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.12);
  box-sizing: border-box;
  border-radius: 4px;
}
.card__body-0-0-22 {
  padding: 1rem;
}
.card__header-0-0-23 {
  padding: 0.5rem 1rem;
  font-size: 0.8rem;
  background: #eee;
  border-bottom: 1px solid #dfdfdf;
}
.card__header-0-0-23 ::slotted(*) {
  margin: 0;
}
.card__footer-0-0-24 {
  padding: 0.5rem 1rem;
  font-size: 0.8rem;
  background: #eee;
  border-top: 1px solid #dfdfdf;
}
.card__footer-0-0-24 ::slotted(*) {
  margin: 0;
}
    </style>
    <div class="card-0-0-21">
      <div class="card__body-0-0-22">
        <slot></slot>
      </div>
      <div class="card__footer-0-0-24">
        <slot name="footer"></slot>
      </div>
    </div>
  </mock:shadow-root>
  <h3 slot="footer">
    Footer content
  </h3>
  A title
</scale-card>
`;

exports[`Card should match snapshot with header slot 1`] = `
<scale-card>
  <mock:shadow-root>
    <style>
      .card-0-0-13 {
  color: #333;
  width: 100%;
  border: 1px solid rgba(0, 0, 0, 0.1);
  overflow: hidden;
  max-width: 400px;
  background: #fff;
  box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.12);
  box-sizing: border-box;
  border-radius: 4px;
}
.card__body-0-0-14 {
  padding: 1rem;
}
.card__header-0-0-15 {
  padding: 0.5rem 1rem;
  font-size: 0.8rem;
  background: #eee;
  border-bottom: 1px solid #dfdfdf;
}
.card__header-0-0-15 ::slotted(*) {
  margin: 0;
}
.card__footer-0-0-16 {
  padding: 0.5rem 1rem;
  font-size: 0.8rem;
  background: #eee;
  border-top: 1px solid #dfdfdf;
}
.card__footer-0-0-16 ::slotted(*) {
  margin: 0;
}
    </style>
    <div class="card-0-0-13">
      <div class="card__header-0-0-15">
        <slot name="header"></slot>
      </div>
      <div class="card__body-0-0-14">
        <slot></slot>
      </div>
    </div>
  </mock:shadow-root>
  <h3 slot="header">
    Header content
  </h3>
  A title
</scale-card>
`;

exports[`Card should match snapshot with image 1`] = `
<scale-card image-top="http://placehold.it/400x300">
  <mock:shadow-root>
    <style>
      .card-0-0-29 {
  color: #333;
  width: 100%;
  border: 1px solid rgba(0, 0, 0, 0.1);
  overflow: hidden;
  max-width: 400px;
  background: #fff;
  box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.12);
  box-sizing: border-box;
  border-radius: 4px;
}
.card__body-0-0-30 {
  padding: 1rem;
}
.card__header-0-0-31 {
  padding: 0.5rem 1rem;
  font-size: 0.8rem;
  background: #eee;
  border-bottom: 1px solid #dfdfdf;
}
.card__header-0-0-31 ::slotted(*) {
  margin: 0;
}
.card__footer-0-0-32 {
  padding: 0.5rem 1rem;
  font-size: 0.8rem;
  background: #eee;
  border-top: 1px solid #dfdfdf;
}
.card__footer-0-0-32 ::slotted(*) {
  margin: 0;
}
    </style>
    <div class="card-0-0-29">
      <img alt="" src="http://placehold.it/400x300">
      <div class="card__body-0-0-30">
        <slot></slot>
      </div>
    </div>
  </mock:shadow-root>
  A title
</scale-card>
`;
